src/plugins/dashboard/public/dashboard_container/component/viewport/dashboard_viewport.tsx

```diff
@@ -6,18 +6,32 @@
  * Side Public License, v 1.
  */
 
-import React, { useEffect, useRef } from 'react';
+import { debounce } from 'lodash';
+import classNames from 'classnames';
+import useResizeObserver from 'use-resize-observer/polyfilled';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 
+import { EuiPortal } from '@elastic/eui';
 import { ViewMode } from '@kbn/embeddable-plugin/public';
 import { ExitFullScreenButton } from '@kbn/shared-ux-button-exit-full-screen';
 
-import { css } from '@emotion/react';
-import { EuiPortal } from '@elastic/eui';
 import { DashboardGrid } from '../grid';
 import { pluginServices } from '../../../services/plugin_services';
 import { DashboardEmptyScreen } from '../empty_screen/dashboard_empty_screen';
 import { useDashboardContainerContext } from '../../dashboard_container_context';
 
+export const useDebouncedWidthObserver = (wait = 250) => {
+  const [width, setWidth] = useState<number>(0);
+  const onWidthCange = useMemo(() => debounce(setWidth, wait), [wait]);
+  const { ref } = useResizeObserver<HTMLDivElement>({
+    onResize: (dimensions) => {
+      if (width === 0) setWidth(dimensions.width);
+      if (dimensions.width !== width) onWidthCange(dimensions.width);
+    },
+  });
+  return { ref, width };
+};
```

Comment:
Debounce the width changes so that resizing the window appears a while lot smoother :+1: